paparazzi/paparazzi/src/main/java/app/cash/paparazzi/internal/resources/ResourceFolderRepository.kt

```diff
@@ -75,4 +87,127 @@ class ResourceFolderRepository(
+    // Use LinkedListMultimap to preserve ordering for editors that show original order.
+    resourceTable.computeIfAbsent(type) { LinkedListMultimap.create() }
+
+  private class Loader(
+    private val repository: ResourceFolderRepository
+  ) : RepositoryLoader<ResourceFolderRepository>(
+    resourceDirectoryOrFile = repository.resourceDir.toPath(),
+    resourceFilesAndFolders = null,
+    namespace = repository.namespace
+  ) {
+    private val resources = EnumMap<ResourceType, ListMultimap<String, ResourceItem>>(ResourceType::class.java)
+    private val sources = mutableMapOf<PathString, ResourceItemSource<BasicResourceItem>>()
+    private val fileResources = hashMapOf<PathString, BasicFileResourceItem>()
+
+    fun load() {
+      if (!resourceDirectoryOrFile.exists()) {
+        return
+      }
+
+      scanResFolder()
+      populateRepository()
+    }
+
+    private fun scanResFolder() {
+      try {
+        // Don't report errors if the resource directory doesn't exist. This happens in some tests.
+        if (CancellableFileIo.notExists(resourceDirectoryOrFile)) {
+          return
+        }
+
+        val shouldParseResourceIds = !loadIdsFromRTxt()
+        val sourceFilesAndFolders =
+          resourceFilesAndFolders?.map { it.toPath()!! } ?: listOf(resourceDirectoryOrFile)
+        for (file in findResourceFiles(sourceFilesAndFolders)) {
+          val folderInfo = file.parentFileName?.let { FolderInfo.create(it, folderConfigCache) }
+          if (folderInfo != null) {
+            // Skip file with the name starting with a dot.
+            if (file.fileName.startsWith(".")) {
+              continue
+            }
+
+            val keyExists = if (folderInfo.folderType == ResourceFolderType.VALUES) {
+              sources.containsKey(file)
+            } else {
+              fileResources.containsKey(file)
+            }
+            if (keyExists) {
+              continue
+            }
+
+            loadResourceFile(file, repository, shouldParseResourceIds)
+          }
+        }
+      } catch (e: Exception) {
+        LOG.severe("Failed to load resources from $resourceDirectoryOrFile: $e")
+      }
+
+      finishLoading(repository)
+
+      // Associate file resources with sources.
+      for ((file, item) in fileResources.entries) {
+        val source: ResourceItemSource<BasicResourceItem> = sources.computeIfAbsent(file) { file ->
+          ResourceItemSourceImpl(file, item.getRepositoryConfiguration())
+        }
+        source.addItem(item)
+      }
+
+      // Populate the resources map.
+      val sortedSources = sources.values.toList()
+      // Sort sources according to folder configurations to have deterministic ordering of resource items in resources.
+      sortedSources.sortedWith(SOURCE_COMPARATOR)
+      for (source in sortedSources) {
+        for (item in source) {
+          getOrCreateMap(item.type).put(item.name, item)
+        }
+      }
+    }
+
+    private fun populateRepository() {
```

Comment:
```suggestion
    private fun populateRepository() {
      repository.sources.putAll(sources)
```

Reply:
`sources` are not used in `repository`, so no need to add it